### ISSUE TITLE
v1.13.1 breaks construction of kafka Consumer when logger provided as a keyword

### ISSUE THREAD

**Issue by rwhit** [DataDog/dd-trace-py#5873]
### Summary of problem

Upgrading from v1.12.1 to v1.13.1 broke construction of kafka Consumer when logger provided as a keyword.

### Which version of dd-trace-py are you using?

v1.13.1 is where we first saw the breakage.

### Which version of pip are you using?

We're using poetry v1.10.0 for dependency management.

### Which libraries and their versions are you using?

<details>
  <summary>`pip freeze`</summary>
  confluent-kafka==1.8.2
</details>

### How can we reproduce your problem?

```
mylogger = logging.getLogger()
mylogger.addHandler(logging.StreamHandler())
consumer = confluent_kafka.Consumer({'bootstrap.servers': 'mybroker.com', 'group.id': 'foo'}, logger=mylogger)
```

### What is the result that you get?

`TypeError: __init__() got an unexpected keyword argument 'logger'`

### What is the result that you expected?

consumer successfully created

**Comment by rwhit** [comment]
I believe this commit is the culprit: https://github.com/DataDog/dd-trace-py/commit/29a884d5c20f4bf195a6f99a01d2f03e110bce41

**Comment by Yun-Kim** [comment]
The fix is merged into 1.x and should be released soon on the 1.12.9, 1.13.4, and 1.14.0 releases.
